protocols/imap/src/main/java/org/apache/james/imap/processor/ListProcessor.java

```diff
@@ -140,9 +153,18 @@ private Mono<Void> respondMailboxList(String referenceName, String mailboxName,
 
         MailboxPath basePath = computeBasePath(session, finalReferencename, isRelative);
 
-        return getMailboxManager().search(mailboxQuery(basePath, mailboxName, mailboxSession), Minimal, mailboxSession)
-            .doOnNext(metaData -> processResult(responder, isRelative, metaData, getMailboxType(session, metaData.getPath())))
-            .then();
+        if (selectSubscribed) {
+            return Flux.from(Throwing.supplier(() -> subscriptionManager.subscriptionsReactive(mailboxSession)).get())
+                .collectList()
+                .flatMapMany(litSubscribed -> getMailboxManager().search(mailboxQuery(basePath, mailboxName, mailboxSession), Minimal, mailboxSession)
+                    .filter(metaData -> litSubscribed.contains(metaData.getPath()))
+                    .doOnNext(metaData -> processResult(responder, isRelative, metaData, getMailboxType(session, metaData.getPath()), true)))
+                .then();
```

Comment:
Are we supposed to also return subscribed items that do not correspond to an existing mailbox?

CF the RFC:

```
   SUBSCRIBED -  causes the LIST command to list subscribed names,
      rather than the existing mailboxes.  This will often be a subset
      of the actual mailboxes.  It's also possible for this list to
      contain the names of mailboxes that don't exist.  In any case, the
      list MUST include exactly those mailbox names that match the
      canonical list pattern and are subscribed to.  This option is
      intended to supplement the LSUB command.
```

Reply:
> Are we supposed to also return subscribed items that do not correspond to an existing mailbox?

Yes, then we have an attribute `NonExistent` in response